.changeset/backend-token-authentication.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-backend': patch
+---
+
+Make identity valid if subject of token is a github-server token
```

Comment:
I don't think this is a GH server token, but a backstage server-2-server auth token.

Probably also good to call out that this requires the token manager to be passed in.